### PR TITLE
Profile Model

### LINKED ISSUE #3: Gender field with enum, Remove email from profile model, Availability as array of string
I make the gender field to have an enum of Male and Female.
I remove the email from the profile model.
I make the availability field as an array of strings

### LINKED ISSUE #4: Fixes availability field to be array of objects
I redesign the availability field to be an array of objects so that my design would look like this:

"availability": [{"day": "Sun", "time": "9:00 - 9:30am"}, {"day": "Mon", "time": "9:30 - 10:00am"}]

### LINKED ISSUE #6: Fix for add non-binary to enum of gender
I added non-binary for individuals that are neither male nor female

### PR DESCRIPTION
**Pull request by sholabioku**
### Create profile model
- list all fields in the profile model 

### Feilds:
- First Name field with the type string and required.
-  Last Name field with the type string and required,
- Description about the user profile with type string and required,
- Availability of the user profile with type boolean and set to default of false,
- Implemented the feedback for the review

### CODE REVIEW

File: server/models/Profile.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  },
+  description: {
+    type: String,
+    required: true,
+  },
+  availabiltity: {
+    type: Boolean,
+    default: false,
+  }
+}, { timestamps: true })
```
Comment (rajivtitus) [inline server/models/Profile.js]: 👍🏽

File: server/models/Profile.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+module.exports = Profile = mongoose.model("profile", profileSchema);
```
Comment (rajivtitus) [inline server/models/Profile.js]:
```suggestion
module.exports = Profile = mongoose.model("Profile", profileSchema);
```
Optional: Please use PascalCase for model names :)

File: server/models/Profile.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  },
+  lastName: {
+    type: String,
+    required: true,
+  },
+  description: {
+    type: String,
+    required: true,
+  },
+  availabiltity: {
```
Comment (rajivtitus) [inline server/models/Profile.js]: Question: Will this design for availability fit our use case for the project?

A dog sitter should have the ability to specify their weekly availability - The days on which they are available to work (Sun-Sat) and the time they are available each day 😄

File: server/models/Profile.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+const mongoose = require('mongoose')
+
+const profileSchema = new mongoose.Schema({
```
Comment (rajivtitus) [inline server/models/Profile.js]: You can also add any additional fields to this model as per our projects' use cases :)

**Review by rajivtitus — COMMENTED** [review]
Really good start! 
- Good use of mongoose
- Nice work adding mongooses' built in validators
- Maybe also consider adding a reference from the User to the Profile model or both ways depending on how you want to query?
I've left some comments for you to have a look and address 😄

File: server/models/Profile.js (hunk 4 above)
Comment (sholabioku) [reply server/models/Profile.js]: Okay

**Comment by rajivtitus** [comment]
Nice work starting to work on those comments, once you've addressed all of them please tag me in for another review!

File: server/models/Profile.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+      required: true,
+    },
+    address: {
+      type: String,
+    },
+    availabiltity: {
+      type: Array,
+      default: [],
+    },
+    user: {
```
Comment (rajivtitus) [inline server/models/Profile.js:41]: 👍🏽

File: server/models/Profile.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    description: {
+      type: String,
+    },
+    gender: {
+      type: String,
+    },
+    birthDate: {
+      type: Date,
+    },
+    email: {
```
Comment (rajivtitus) [inline server/models/Profile.js]: Please remove the email field - it already exists in the User model and we do not want to store duplicate data across our DB :)

File: server/models/Profile.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    },
+    lastName: {
+      type: String,
+      required: true,
+    },
+    description: {
+      type: String,
+    },
+    gender: {
+      type: String,
```
Comment (rajivtitus) [inline server/models/Profile.js:17]: Optional: You can look into using enum validators for the gender field

Ref: https://mongoosejs.com/docs/validation.html

File: server/models/Profile.js [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+      type: Date,
+    },
+    email: {
+      type: String,
+      required: true,
+    },
+    address: {
+      type: String,
+    },
+    availabiltity: {
```
Comment (rajivtitus) [inline server/models/Profile.js]: Please give me an example of how this design for availability will work? :)

**Review by rajivtitus — COMMENTED** [review]
Thanks for adding those additional fields, please have a look at a couple of comments I've just made. Thanks

File: server/models/Profile.js (hunk 8 above)
Comment (sholabioku) [reply server/models/Profile.js]: I am trying to set an array that contains a list of objects that each would key for dates and days which the user will be available:

Comment (sholabioku) [reply server/models/Profile.js]: Example
["Sun 2am-4pm", "Mon 4am-6am"]

File: server/models/Profile.js [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+    lastName: {
+      type: String,
+      required: true,
+    },
+    description: {
+      type: String,
+    },
+    gender: {
+      type: String,
+      enum: ['male', 'female'],
```
Comment (rajivtitus) [inline server/models/Profile.js]:
```suggestion
      enum: ['male', 'female', 'non-binary'],
```
For individuals who don't identify as male/female 😄

File: server/models/Profile.js (hunk 8 above)
Comment (rajivtitus) [reply server/models/Profile.js]: Thanks for the clarification, the UI will have to be designed accordingly 👍🏽

**Review by rajivtitus — APPROVED** [review]
Good work making the requested changes! Minor comment to address before merging. Thanks :)

**Pull request merged**